yt_dlp/extractor/radiocomercial.py

```diff
@@ -0,0 +1,137 @@
+import functools
+
+from .common import InfoExtractor
+from ..utils import (
+    OnDemandPagedList,
+    extract_attributes,
+    get_element_by_class,
+    get_element_html_by_class,
+    get_element_text_and_html_by_tag,
+    get_elements_html_by_class,
+    int_or_none,
+    join_nonempty,
+    try_call,
+    unified_strdate,
+    urljoin
+)
+from ..utils.traversal import traverse_obj
+
+
+class RadioComercialIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?radiocomercial\.pt/podcasts/[^/]+/\D*(?P<season>\d+)/(?P<id>[\w-]+)'
```

Comment:
There appears to be at least one exception: https://radiocomercial.pt/podcasts/tnt-todos-no-top
This one is using the year, instead of `t<value>` for the newer podcasts

Reply:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?radiocomercial\.pt/podcasts/[^/?#]+/t?(?P<season>\d+)/(?P<id>[\w-]+)'
```